reporter/src/main/kotlin/reporters/AntennaAttributionDocumentReporter.kt

```diff
@@ -132,36 +110,73 @@ class AntennaAttributionDocumentReporter : Reporter {
             }
         }
 
-        val generator = AttributionDocumentGeneratorImpl(
-            reportFilename,
-            workingDir,
-            templateId,
-            DocumentValues(rootProject.id.name, rootProject.id.version, projectCopyright)
-        )
+        val outputFiles = mutableListOf<File>()
+        val projects = input.ortResult.getProjects(omitExcluded = true)
+
+        try {
+            projects.forEach { project ->
+                val dependencies = project.collectDependencies()
 
-        val documentFile = if (templateFiles.size == templateFileNames.size) {
-            generator.generate(
-                    artifacts,
-                    templateFiles[TEMPLATE_COVER_PDF],
-                    templateFiles[TEMPLATE_COPYRIGHT_PDF],
-                    templateFiles[TEMPLATE_CONTENT_PDF],
-                    templateFiles[TEMPLATE_BACK_PDF]
+                val packages = input.ortResult.getPackages().mapNotNull { (pkg, _) ->
+                    pkg.takeIf { it.id in dependencies }
+                }
+
+                val artifacts = packages.map { pkg ->
+                    val licenses = collectLicenses(pkg.id, input.ortResult)
+                    AttributionDocumentPdfModel(
+                        purl = pkg.purl,
+                        binaryFilename = pkg.binaryArtifact.takeUnless { it.url.isEmpty() }?.let { File(it.url).name },
+                        declaredLicenses = licenses.map { createLicenseInfo(it, input.licenseTextProvider) },
+                        copyrightStatements = createCopyrightStatement(pkg.id, licenses, licenseFindings)
+                    )
+                }.toList()
+
+                val projectCopyright = createCopyrightStatement(
+                    project.id,
+                    collectLicenses(project.id, input.ortResult),
+                    licenseFindings
                 )
-        } else {
-            try {
-                generator.generate(artifacts)
-            } finally {
-                if (templateId != DEFAULT_TEMPLATE_ID) removeTemplateFromClassPath()
+
+                val workingDir = createTempDir()
+
+                val generator = AttributionDocumentGeneratorImpl(
+                    reportFilename,
+                    workingDir,
+                    templateId,
+                    DocumentValues(project.id.name, project.id.version, projectCopyright)
+                )
+
+                val documentFile = if (templateFiles.size == templateFileNames.size) {
+                    generator.generate(
+                        artifacts,
+                        templateFiles[TEMPLATE_COVER_PDF],
+                        templateFiles[TEMPLATE_COPYRIGHT_PDF],
+                        templateFiles[TEMPLATE_CONTENT_PDF],
+                        templateFiles[TEMPLATE_BACK_PDF]
+                    )
+                } else {
+                    try {
```

Comment:
The `try` / `finally` should be dropped here at it's already (correctly) done in the end.